tools/run_tests/run_xds_tests.py

```diff
@@ -3274,18 +3270,18 @@ def __init__(self, compute, alpha_compute, project, project_num):
                     logger.info('Client output:')
                     with open(test_log_filename, 'r') as client_output:
                         logger.info(client_output.read())
-        if not os.path.exists(_TEST_LOG_BASE_DIR):
-            os.makedirs(_TEST_LOG_BASE_DIR)
-        report_utils.render_junit_xml_report(test_results,
-                                             os.path.join(
-                                                 _TEST_LOG_BASE_DIR,
-                                                 _SPONGE_XML_NAME),
-                                             suite_name='xds_tests',
-                                             multi_target=True)
+
         if failed_tests:
             logger.error('Test case(s) %s failed', failed_tests)
             sys.exit(1)
```

Comment:
Do we want all failed runs to be saved? If GCP has an outage, our quota might run out.

Reply:
Maybe I'm missing something, but I think this only saves resources if `skip_cleanup_to_debug_failure` is set to true, which only happens when the load_report test fails. The changed line here just moves the call to `render_junit_xml_report` to the finally block.

Reply:
Also, for what it's worth, this change should only need to be live for ~1 day before we "catch" a failing instance of the load report test, and until then I'll just be manually babysitting the project (aka, running the cleanup script) to make sure we don't save too many sets of backends.

Reply:
TIL again that `sys.exit` raises `SystemExit`, it's not a hard halting at all. LGTM.